### PR TITLE
fix: issue #59 — ad-hoc 微小正定数の整理と論文外モデリング床の削除

### LINKED ISSUE #59: Follow-up: 正則化・安定化のための ad-hoc 微小正定数の要否をPR  54〜58 マージ後に再検証する
現状（PR #54 の時点）では `tda_ml/topology.py` 内の `compute_anisotropic_metric`・`compute_anisotropic_distance_matrix` で軸長パラメータの数値ハック（`+1e-6`, `+1e-8` などの正則化）が入っている。だが、これらのパラメータ調整が本当に必要かどうかは、今後この関数群がどのように使われるか（後続のPR群や利用状況）による。

現時点では:
- テストコードからしか使われていない
- 本格的な用途で呼ばれるのは今後のPRのマージ以降になる
- 数値ハックを残すべき根拠が薄い状態

よって、**全ての関連PR（モデル・損失・トレーナ等含む）がマージされた段階**で一度「これらの `+1e-6`/`+1e-8` 等の正則化パラメータが本当に必要か」を再検証し、必要性が薄ければ削除／設計見直しを検討すること。

- 一時的な対策として入れたものなら追跡必須
- 実運用バグ・分岐条件クリア不能など現象として現れるか注視する
- 必要性判明時は理由・証拠も追記のこと

**後続PR #55 #56 #57 #58 が全て入りきった段階で必ず確認タスクとして再検討するためのリマインダーです。**

### PR DESCRIPTION
**Pull request by koki3070**
## Summary

Closes #59.

PR #54–#58 マージ後の follow-up として、教師あり本線 `tda_ml/` の ad-hoc 微小正定数を棚卸しし、論文に無いモデリング床を削除しました。再検証で見つかった `sqrt(0)` 勾配爆発（AMP 下で学習停止）も修正しています。

- **削除:** encoder `clamp(base_axes, min=0.2)`、legacy `axes + 1e-4`、metric への軸長 `+1e-6`
- **集約:** `tda_ml/numerical_eps.py`（`NUMERICAL_EPS`, `PCA_RIDGE_EPS`, `EIGENVALUE_FLOOR`, `INLIER_PROB_MIN`）
- **修正:** `sqrt(dist_sq + NUMERICAL_EPS)`（対角の勾配有限化；宣言的 L2 床）
- **厳格化:** `TopologicalLoss` がバッチ全件失敗時に黙って `0` を返さず `RuntimeError`

## Verification

mahalanobis 12ep（`configs/issue59_verify_mahalanobis.yaml`, seed=42, `reproduce_1week_tuned` 相当）:

| 指標 | 値 |
|------|-----|
| best val_mcc | 0.7640 |
| test_mcc | 0.751 |
| run | `outputs/issue59_verify_mahalanobis_20260617_090007` |

詳細は [issue コメント](https://github.com/uda-lab/TDA-ML/issues/59#issuecomment-4724915096) を参照。

## Test plan

- [x] `pytest tests/test_topology.py tests/test_topology_gradients.py`
- [x] `test_legacy_axes_positive_without_modeling_floor`
- [x] 教師あり 12ep mahalanobis 再検証（上記）

## Out of scope (follow-up)

- 退化ケースの明示分岐設計
- `rk_tanh` / `hybrid_rk_tanh` 等の軸パラメータ拡張
- 教師あり本線外（`reproduce_pd_animation.py` 等）の定数棚卸し

Made with [Cursor](https://cursor.com)

### CODE REVIEW

**Comment by t-uda** [comment]
@codex review this PR

File: tda_ml/topology.py [hunk 1]
```diff
@@ -122,5 +124,6 @@ def compute_anisotropic_distance_matrix(
             f"symmetrize must be 'max' or 'min', got {symmetrize!r}"
         )
 
-    dist_sq = torch.clamp(dist_sq, min=0.0)
-    return torch.sqrt(dist_sq)
+    # sqrt'(0) is undefined; diagonal entries are exactly zero, so add NUMERICAL_EPS
+    # inside sqrt for a bounded derivative (declared L2 floor, not a modeling prior).
+    return torch.sqrt(dist_sq + NUMERICAL_EPS)
```
[the remark this answers is not in the document]
Comment (t-uda) [reply tda_ml/topology.py:129]: 却下．不要な clamp

**Comment by koki3070** [comment]
## レビュー指摘への対応（`82f9528`）

### Codex P2 — `REPRODUCIBILITY.md` / legacy `+1e-4`（対応済み）
- `AnisotropicOutlierClassifier.forward` の `axes = axes_scale * base_axes + 1e-4` を削除し、`axes = axes_scale * base_axes` に統一
- コミットメッセージ・`REPRODUCIBILITY.md`・`test_legacy_axes_positive_without_modeling_floor` の記述と実装を一致

### Codex P2 — `experiments/issue59_verify_mahalanobis.py` 欠落（対応済み）
- `experiments/issue59_verify_mahalanobis.py` を追加（`configs/issue59_verify_mahalanobis.yaml` を読み、preflight 後に `tda_ml.main` を実行）
- `tda_ml/supervised_diagnostics.py` と `tda_ml/main.py` の `early_abort` / `run_manifest.json` / abort レポート出力を統合
- `configs/issue59_verify_mahalanobis.yaml` で `early_abort.enabled: true`（`probe_epochs: 8`）
- `tests/test_supervised_diagnostics.py` 追加
- `REPRODUCIBILITY.md` のリポジトリ範囲・`probe_epochs` 説明を更新

### Codex P2 — `tda_ml/topology.py` の clamp 復帰提案（対応なし）
- t-uda さんコメント「却下．不要な clamp」に従い、`sqrt(dist_sq + NUMERICAL_EPS)` のまま変更していません

### 検証
- `uv run pytest` パス
- CI（Ruff workflow: ruff + pytest + repro-smoke）パス

**Review by t-uda — APPROVED** [review]
LGTM
